UndertaleModLib/Util/TextureWorker.cs

```diff
@@ -1,217 +1,250 @@
+
+                return newImage;
             }
         }
 
-        public void ExportAsPNG(UndertaleTexturePageItem texPageItem, string FullPath, string imageName = null, bool includePadding = false)
+        /// <summary>
+        /// Exports the given texture page item to disk, as a PNG, to the supplied path. (With or without padding.)
+        /// </summary>
+        public void ExportAsPNG(UndertaleTexturePageItem texPageItem, string fullPath, string imageName = null, bool includePadding = false)
         {
-            SaveImageToFile(FullPath, GetTextureFor(texPageItem, imageName != null ? imageName : Path.GetFileNameWithoutExtension(FullPath), includePadding));
+            using var image = GetTextureFor(texPageItem, imageName ?? Path.GetFileNameWithoutExtension(fullPath), includePadding);
+            SaveImageToFile(image, fullPath);
         }
 
-        public Bitmap GetTextureFor(UndertaleTexturePageItem texPageItem, string imageName, bool includePadding = false)
+        /// <summary>
+        /// Creates an image representing the sole texture page item supplied, with or without padding.
+        /// </summary>
+        public IMagickImage<byte> GetTextureFor(UndertaleTexturePageItem texPageItem, string imageName, bool includePadding = false)
         {
+            // Get texture page that the item lives on
+            MagickImage embeddedImage = GetEmbeddedTexture(texPageItem.TexturePage);
+
+            // Ensure texture is no larger than its bounding box
             int exportWidth = texPageItem.BoundingWidth; // sprite.Width
             int exportHeight = texPageItem.BoundingHeight; // sprite.Height
-            Bitmap embeddedImage = GetEmbeddedTexture(texPageItem.TexturePage);
-
-            // Sanity checks.
-            if (includePadding && ((texPageItem.TargetWidth > exportWidth) || (texPageItem.TargetHeight > exportHeight)))
-                throw new InvalidDataException(imageName + "'s texture is larger than its bounding box!");
+            if (includePadding && (texPageItem.TargetWidth > exportWidth || texPageItem.TargetHeight > exportHeight))
+            {
+                throw new InvalidDataException($"{imageName}'s texture is larger than its bounding box!");
+            }
 
-            // Create a bitmap representing that part of the texture page.
-            Bitmap resultImage = null;
+            // Create an image cropped from the item's part of the texture page
+            IMagickImage<byte> resultImage = null;
             lock (embeddedImage)
             {
                 try
                 {
-                    resultImage = embeddedImage.Clone(new Rectangle(texPageItem.SourceX, texPageItem.SourceY, texPageItem.SourceWidth, texPageItem.SourceHeight), PixelFormat.DontCare);
+                    resultImage = embeddedImage.Clone(texPageItem.SourceX, texPageItem.SourceY, texPageItem.SourceWidth, texPageItem.SourceHeight);
                 }
                 catch (OutOfMemoryException)
```

Comment:
Honestly I have no idea at this point. I'm personally fine with removing this try..catch altogether, especially since the underlying code is different now. I'm assuming this was a hack added to account for some kind of overflow. What do you think?

Reply:
considering you have a few clones further down below, id just remove it yeah